### PR TITLE
Add RBS files for DelayedJob ActiveRecord Backend(4.1) gem.

### PR DESCRIPTION
**Pull request by M-Yamashita01**
## Description
Complete RBS write-up of [DelayedJob ActiveRecord Backend](https://github.com/collectiveidea/delayed_job_active_record)(4.1)
- Added RBS files for DelayedJob ActiveRecord Backend (4.1)
- Added test script in _scripts folder
- Added tests and Steepfile in _test folder
- Added git module referencing the source code of the gem

### CODE REVIEW

File: gems/delayed_job_active_record/4.1/active_record.rbs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+      class Job
+        def self.set_delayed_job_table_name: -> String
+        def self.ready_to_run: (untyped worker_name, untyped max_run_time) -> untyped
+        def self.before_fork: -> untyped
+        def self.after_fork: -> untyped
+        def self.clear_locks!: (untyped worker_name) -> untyped
+        def self.reserve: (untyped worker, ?untyped max_run_time) -> nil
+        def self.reserve_with_scope: (untyped ready_scope, untyped worker, Time now) -> nil
+        def self.reserve_with_scope_using_optimized_sql: (untyped ready_scope, untyped worker, Time now) -> nil
```
Comment (pocke) [inline gems/delayed_job_active_record/4.1/active_record.rbs:23]: They, `reserve_with_scope_using_optimized_mysql` and `reserve_with_scope_using_optimized_mssql` probably return non-nil value. I guess `Job?` is the correct returned type, please confirm that.

File: gems/delayed_job_active_record/4.1/active_record.rbs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+module Delayed
+  module Backend
+    module ActiveRecord
+      self.@configuration: Configuration
+
+      def self.configuration: -> Configuration
+      def self.configure: -> untyped
+
+      class Configuration
+        attr_reader reserve_sql_strategy: :optimized_sql
```
[the remark this answers is not in the document]
Comment (M-Yamashita01) [reply gems/delayed_job_active_record/4.1/active_record.rbs]: Thank you.
We have set the `Symbol`.

File: gems/delayed_job_active_record/4.1/active_record.rbs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+module Delayed
+  module Backend
+    module ActiveRecord
+      self.@configuration: Configuration
+
+      def self.configuration: -> Configuration
+      def self.configure: -> untyped
```
[the remark this answers is not in the document]
Comment (M-Yamashita01) [reply gems/delayed_job_active_record/4.1/active_record.rbs]: Thank you.
We added block types and return values.

File: gems/delayed_job_active_record/4.1/active_record.rbs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+      class Configuration
+        attr_reader reserve_sql_strategy: :optimized_sql
+        def initialize: -> void
+        def reserve_sql_strategy=: (Symbol val) -> Symbol
+      end
+
+      class Job
```
[the remark this answers is not in the document]
Comment (M-Yamashita01) [reply gems/delayed_job_active_record/4.1/active_record.rbs]: Thank you.
We added library in the Steepfile and fixed arguments in the test script.

File: gems/delayed_job_active_record/4.1/active_record.rbs (hunk 1 above)
Comment (M-Yamashita01) [reply gems/delayed_job_active_record/4.1/active_record.rbs]: That is as you pointed out.
These methods return the value of `Job?`
We fixed them.

**Review by pocke — APPROVED** [review]
Thanks!

**Comment by pocke** [comment]
By the way, please avoid force pushing if it isn't required. Force pushing conceals the diff between the commit I reviewed previously and the HEAD, so I need to review the entire of your PR. It wastes the reviewer's time.

**Comment by M-Yamashita01** [comment]
Thank you for pointing this out.
We apologize for wasting the reviewer's time.
We thought it should be one commit per pull request, so we did a `git rebase`.
We will refrain from using rebase in the next pull request.

**Pull request merged**